lib/Check.vala

```diff
@@ -89,4 +89,20 @@ public abstract class ValaLint.Check : Object {
             critical ("%s is not a valid Regex", pattern);
         }
     }
+
+    /**
+     * Adds a mistake to the mistake list and checks for duplicates.
+     *
+     * @param mistake The mistake.
+     * @param mistakes The mistakes list.
+     */
+    protected void add_mistake (FormatMistake mistake, ref Vala.ArrayList<FormatMistake?> mistakes) {
+        mistakes.equal_func = (a, b) => {
+            return a == b;
+        };
```

Comment:
This seems an odd place to add an equal_func - should this not have been set when the list was created?

Reply:
This should be fixed with the latest commits.